NewsApp/NewsApp/Model/APIClient.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  APIClient.swift
+//  NewsApp
+//
+//  Created by Josselin DUBOC on 23/03/2020.
+//  Copyright © 2020 Josselin DUBOC. All rights reserved.
+//
+
+import Foundation
+
+public class APIClient {
+    private let baseEndpointUrl = URL(string: "http://newsapi.org/v2/")!
+    private let session = URLSession(configuration: .default)
+    public var newsArticles: [Article] = []
+
+    public func send<T: APIRequest>(_ request: T, completion: @escaping (Result<[Article], Error>) -> Void) {
+        let endpoint = self.endpoint(for: request)
+
+        let task = session.dataTask(with: URLRequest(url: endpoint)) { data, _, error in
+            if let data = data {
+                do {
+                    // Decode the top level response, and look up the decoded response to see
+                    // if it's a success or a failure
+                    let newsResponse = try JSONDecoder().decode(NewsResponse<T.Response>.self, from: data)
+                    if newsResponse.status == "error" {
+                        completion(.failure(NewsError.server(message: newsResponse.code ?? newsResponse.error ?? "Erreur dans votre requête")))
+                    } else if newsResponse.status == "ok" {
```

Comment:
Dès que tu vois une string, un test sur une string, tu peux te demander est-ce que je pourrais utiliser une enum ?
Ici je pense que ca sera plus "safe" !

Reply:
fix in "fix PR feature/topFR"